src/cpr_data_access/models.py

```diff
@@ -1097,3 +1105,105 @@ def to_huggingface(
         )
 
         return huggingface_dataset
+
+    def from_huggingface(
+        self,
+        huggingface_dataset: HFDataset,
+        limit: Optional[int] = None,
+    ) -> "Dataset":
+        """
+        Create a dataset from a huggingface dataset.
+
+        :param huggingface_dataset: created using `Dataset.to_huggingface()`
+        :param limit: optionally limit the number of documents to load
+        :return self: with documents loaded from huggingface dataset
+        """
+
+        document_metadata_model = self.document_model.__fields__[
+            "document_metadata"
+        ].outer_type_
```

Comment:
Why is the `__fields__` bit needed here?

Reply:
`__fields__` gives you the field names of the pydantic model. Acknowledged this is a bit strange if you're not familiar with pydantic – I've added a `get_document_metadata_model` classmethod to use here instead.

Reply:
actually, removing this as it's not longer needed once i'd addressed your comment below.